bach/tests/functional/bach/test_df_fillna.py

```diff
@@ -91,6 +99,29 @@ def test_fillna_w_methods_against_pandas(engine) -> None:
     )
 
 
+def test_ffill_propagation(engine) -> None:
+    pdf = pd.DataFrame(DATA, columns=list("ABCDEFG"))
+    pdf['groups'] = pdf.B>0
+    df = DataFrame.from_pandas(engine=engine, df=pdf, convert_objects=True).reset_index().sort_values('_index_0')
+    pdf = pdf.reset_index(names='_index_0').sort_values('_index_0')
```

Comment:
Shall we change it to this?
```suggestion
    pdf = pdf.reset_index().rename(columns={'index': '_index_0'})
```